src/dev/lupio/lupio_rtc.cc

```diff
@@ -0,0 +1,103 @@
+#include "dev/lupio/lupio_rtc.hh"
+
+#include <ctime>
+#include <string>
+
+#include "base/bitfield.hh"
+#include "base/time.hh"
+#include "base/trace.hh"
+#include "debug/LupioRTC.hh"
+#include "mem/packet.hh"
+#include "mem/packet_access.hh"
+#include "params/LupioRTC.hh"
+#include "sim/system.hh"
+
+namespace gem5
+{
+
+using namespace RiscvISA;
+
+LupioRTC::LupioRTC(const Params &params) :
+    BasicPioDevice(params, params.pio_size),
+    system(params.system),
+    time(params.time)
+{
+    struct timeval tv;
+    struct tm time_bd;
+
+    gettimeofday(&tv, NULL);
+    gmtime_r(&tv.tv_sec, &time_bd);
+
+    DPRINTFN("Real-time clock set to %s", asctime(&time_bd));
+}
+
+LupioRTC::~LupioRTC()
+{
+}
+
+uint8_t
+LupioRTC::lupioRTCRead(uint8_t addr)
+{
+    uint8_t r = 0;
+    struct timeval tv;
+    struct tm time_bd;
+
+    gettimeofday(&tv, NULL);
+    gmtime_r(&tv.tv_sec, &time_bd);
+
+    switch (addr) {
+        case LUPIO_RTC_SECD:
+            r = time_bd.tm_sec;
+            break;
+        case LUPIO_RTC_MINT:
+            r = time_bd.tm_min;
+            break;
+        case LUPIO_RTC_HOUR:
+            r = time_bd.tm_hour;
+            break;
+        case LUPIO_RTC_DYMO:
+            r = time_bd.tm_mday;
+            break;
+        case LUPIO_RTC_MNTH:
+            r = time_bd.tm_mon + 1;
+            break;
+        case LUPIO_RTC_YEAR:
+            r = (time_bd.tm_year + 1900) % 100;
+            break;
+        case LUPIO_RTC_CENT:
+            r = (time_bd.tm_year + 1900) / 100;
+            break;
+        case LUPIO_RTC_DYWK:
+            r = 1 + (time_bd.tm_wday + 6) % 7;
+            break;
+        case LUPIO_RTC_DYYR:
+            r = time_bd.tm_yday + 1;
+            break;
+    }
```

Comment:
Make sure to keep the same comments as the QEMU model to explain the format of each field. Otherwise, it looks a bit random.